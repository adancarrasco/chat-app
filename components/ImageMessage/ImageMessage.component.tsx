import React, { useState } from "react";
import { Animated, Image, StyleSheet } from "react-native";

export default function ImageMessage({ isLocalUser, shouldAnimate }) {
  const { localUser, remoteUser } = styles;
  const [fadeAnim] = useState(new Animated.Value(0));

  React.useEffect(() => {
    Animated.timing(fadeAnim, {
      toValue: 1,
      duration: 700
    }).start();
  }, []);
  return (
    <Animated.View
      style={[
        { opacity: 1 },
        styles.container,
        isLocalUser ? localUser : remoteUser,
        shouldAnimate ? { opacity: fadeAnim } : ""
      ]}
    >
      <Image
        resizeMode="contain"
        style={{ height: 120, width: 180 }}
        source={{
          uri:
            "https://www.petmd.com/sites/default/files/CANS_dogsmiling_379727605.jpg"
        }}
      />
    </Animated.View>
  );
}

const styles = StyleSheet.create({
  container: {
    // flexDirection: "column",
    padding: 10,
    borderRadius: 8,
    // color: "#f00",
    marginBottom: 14
    // width: "65%"
  },
  localUser: {
    backgroundColor: "rgb(140,149,154)"
  },
  remoteUser: {
    backgroundColor: "rgb(43,114,221)",
    alignItems: "flex-end",
    color: "#fff",
    alignSelf: "flex-end"
  },
  userNameText: {
    color: "#fff",
    marginBottom: 6,
    fontWeight: "600"
  },
  messageText: {
    color: "#fff"
  }
});
