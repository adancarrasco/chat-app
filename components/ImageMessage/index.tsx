export { default } from "./ImageMessage.component";
