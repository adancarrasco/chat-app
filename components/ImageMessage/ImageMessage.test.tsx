import React from "react";
import renderer from "react-test-renderer";
import ImageMessage from "./";

describe("<ImageMessage /> rendering", () => {
  const mockMessage = {
    isLocalUser: true,
    userName: "Summer",
    message: "Sorry I cannot call you now.",
    shouldAnimate: false
  };

  it("Matches snapshot as localUser", () => {
    const tree = renderer.create(<ImageMessage {...mockMessage} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Matches snapshot as remoteUser", () => {
    const tree = renderer
      .create(<ImageMessage {...mockMessage} isLocalUser={false} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
