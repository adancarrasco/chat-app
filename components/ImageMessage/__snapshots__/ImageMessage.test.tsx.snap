// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ImageMessage /> rendering Matches snapshot as localUser 1`] = `
<View
  style={
    Object {
      "backgroundColor": "rgb(140,149,154)",
      "borderRadius": 8,
      "marginBottom": 14,
      "opacity": 1,
      "padding": 10,
    }
  }
>
  <Image
    resizeMode="contain"
    source={
      Object {
        "uri": "https://www.petmd.com/sites/default/files/CANS_dogsmiling_379727605.jpg",
      }
    }
    style={
      Object {
        "height": 120,
        "width": 180,
      }
    }
  />
</View>
`;

exports[`<ImageMessage /> rendering Matches snapshot as remoteUser 1`] = `
<View
  style={
    Object {
      "alignItems": "flex-end",
      "alignSelf": "flex-end",
      "backgroundColor": "rgb(43,114,221)",
      "borderRadius": 8,
      "color": "#fff",
      "marginBottom": 14,
      "opacity": 1,
      "padding": 10,
    }
  }
>
  <Image
    resizeMode="contain"
    source={
      Object {
        "uri": "https://www.petmd.com/sites/default/files/CANS_dogsmiling_379727605.jpg",
      }
    }
    style={
      Object {
        "height": 120,
        "width": 180,
      }
    }
  />
</View>
`;
